### PR TITLE
Upgrade flake8 to 3.6.0

### PR DESCRIPTION
**Pull request by ssbarnea**
#### PR Type

- Bugfix Pull Request

Upgrade flake8 in order to avoid undesired warning inside pycodestyle.
See https://github.com/PyCQA/pycodestyle/issues/728

### CODE REVIEW

**Comment by webknjaz** [comment]
@ssbarnea please fill in PR template when creating them. I bet it's not 3 types in one :)

P.S. I've been thinking to do such upgrade in that CI PR but reverted this change as it must be separate. 
P.P.S. I would really want to leverage robots power to prevent humans from being distracted with such boring things as keeping versions up-to-date...
cc @gundalow ^

**Comment by webknjaz** [comment]
Hey @ssbarnea, please rebase this PR.

File: .travis.yml [hunk 1]
```diff
@@ -265,7 +265,7 @@ before_install:
 - gem install inspec rubocop
 - export TOXENV=$(echo $TOXENV_TMPL | envsubst)
 install:
-- pip install tox-venv
+- pip install tox-venv tox-pip-extensions
```
Comment (webknjaz) [inline .travis.yml]: This change is unrelated

File: molecule/logger.py [hunk 2]
```diff
@@ -19,11 +19,27 @@
...
 
 import logging
+import os
 import sys
 
 import colorama
 
-colorama.init(autoreset=True)
+
+def should_do_markup(file=sys.stdout):
```
Comment (webknjaz) [inline molecule/logger.py]: These changes are unrelated.

**Comment by themr0c** [comment]
Linting with this new version of flake8 is throwing new errors that need to be fixed first -- see #1696

```
./molecule/shell.py:57: [W504] line break after binary operator
./molecule/shell.py:67: [W504] line break after binary operator
./molecule/driver/base.py:139: [W504] line break after binary operator
./test/functional/docker/test_scenarios.py:352: [W605] invalid escape sequence '\['
./test/functional/docker/test_scenarios.py:352: [W605] invalid escape sequence '\]'
./test/functional/docker/test_scenarios.py:352: [W605] invalid escape sequence '\['
./test/functional/docker/test_scenarios.py:352: [W605] invalid escape sequence '\]'
./test/functional/docker/test_scenarios.py:353: [W605] invalid escape sequence '\['
./test/functional/docker/test_scenarios.py:353: [W605] invalid escape sequence '\]'
./test/functional/docker/test_scenarios.py:353: [W605] invalid escape sequence '\['
./test/functional/docker/test_scenarios.py:353: [W605] invalid escape sequence '\]'
./test/functional/docker/test_scenarios.py:354: [W605] invalid escape sequence '\['
./test/functional/docker/test_scenarios.py:354: [W605] invalid escape sequence '\]'
./test/functional/docker/test_scenarios.py:354: [W605] invalid escape sequence '\['
./test/functional/docker/test_scenarios.py:354: [W605] invalid escape sequence '\]'
```

**Comment by gundalow** [comment]
@ssbarnea I believe this was done by https://github.com/ansible/molecule/pull/1695 so closing
